src/aco.cpp

```diff
@@ -196,10 +206,50 @@ namespace full_coverage_path_planner
                         }
                     }
 
+                    //move ant
                     visited[new_location.pos.y][new_location.pos.x] = eNodeVisited;
                     ant.move(new_location, true);
                     ant.visited_counter++;
                     goals = map_2_goals(visited, eNodeOpen);
+
+                    //move ant extra according to velocity
+                    int moved = 1;
+                    int nx = new_location.pos.x;
+                    int ny = new_location.pos.y;
+
+                    while (moved != ant.velocity)
+                    {
+                        if (ant.direction == NORTH)
+                        {
+                            ny++;
+                        }
+                        if (ant.direction == EAST)
+                        {
+                            nx++;
+                        }
+                        if (ant.direction == SOUTH)
+                        {
+                            ny--;
+                        }
+                        if (ant.direction == WEST)
+                        {
+                            nx--;
+                        }
+
+                        gridNode_t nPos = {{nx, ny}, 0, 0};
+
+                        if(visited[ny][nx] == eNodeOpen)
+                        {
+                            ant.move(nPos, true);
+                            ant.visited_counter++;
+                            visited[ny][nx] = eNodeVisited;
+                            moved++;
+                        }
```

Comment:
Generate a new position from the increments, and add these increments to the ant path.

ONLY if the following conditions are met:
-The next tile is not part of the ants working memory (already visited tiles)
-The next tile is not blocked.

Increment the moved counter, and keep repeating this in a while loop until the moved counter matches the ant velocity.